tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs

```diff
@@ -606,15 +612,89 @@ private void WindowEx_WindowStateChanged(object sender, WindowState e)
     {
         if (e.Equals(WindowState.Normal))
         {
-            if (Height == FloatingHorizontalBarHeight)
+            // If, as part of being restored, we were in an expanded state, then make our window bigger (don't go back to collapsed mode)
+            if (_viewModel.ShowingExpandedContent && Height == FloatingHorizontalBarHeight)
+            {
+                Height = Settings.Default.ExpandedWindowHeight;
+            }
+        }
+        else if (e.Equals(WindowState.Maximized))
+        {
+            // If we're being maximized, expand our content
+            _viewModel.ShowingExpandedContent = true;
+        }
+    }
+
+    private Windows.Win32.UI.Shell.SUBCLASSPROC? _wndProc;

```

Comment:
>Windows.Win32.UI.Shell. [](http://example.com/codeflow?start=12&length=23)

nit: using unless this is for disambiguation

Reply:
Updated.